File: samples/client/petstore/csharp-dotnet2/SwaggerClientTest/Lib/SwaggerClient/src/main/CsharpDotNet2/IO/Swagger/Model/SubCategory.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class SubCategory {
    /// <summary>
    /// Gets or Sets Category
    /// </summary>
    [DataMember(Name="category", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "category")]
    public AllOfSubCategoryCategory Category { get; set; }

    /// <summary>
    /// Gets or Sets Category2
    /// </summary>
    [DataMember(Name="category2", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "category2")]
    public Category Category2 { get; set; }

    /// <summary>
    /// Gets or Sets Pets
    /// </summary>
    [DataMember(Name="pets", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "pets")]
    public List<AllOfSubCategoryPetsItems> Pets { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class SubCategory {\n");
      sb.Append("  Category: ").Append(Category).Append("\n");
      sb.Append("  Category2: ").Append(Category2).Append("\n");
      sb.Append("  Pets: ").Append(Pets).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: samples/client/petstore/dart/swagger/lib/model/sub_category.dart
part of swagger.api;

class SubCategory {
  
  AllOfSubCategoryCategory category = null;

  Category category2 = null;

  List<SubCategoryPets> pets = [];

  SubCategory();

  @override
  String toString() {
    return 'SubCategory[category=$category, category2=$category2, pets=$pets, ]';
  }

  SubCategory.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    category = new AllOfSubCategoryCategory.fromJson(json['category']);
    category2 = new Category.fromJson(json['category2']);
    pets = SubCategoryPets.listFromJson(json['pets']);
  }

  Map<String, dynamic> toJson() {
    return {
      'category': category,
      'category2': category2,
      'pets': pets
     };
  }

  static List<SubCategory> listFromJson(List<dynamic> json) {
    return json == null ? new List<SubCategory>() : json.map((value) => new SubCategory.fromJson(value)).toList();
  }

  static Map<String, SubCategory> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, SubCategory>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new SubCategory.fromJson(value));
    }
    return map;
  }
}
